packages/autocomplete/src/elements/Autocomplete.js

```diff
@@ -0,0 +1,270 @@
+    'aria-label': PropTypes.string,
+    validation: PropTypes.oneOf([
+      VALIDATION.SUCCESS,
+      VALIDATION.WARNING,
+      VALIDATION.ERROR,
+      VALIDATION.NONE
+    ]),
+    message: PropTypes.node,
+    hint: PropTypes.node,
+    small: PropTypes.bool,
+    selectedValue: PropTypes.string,
+    onChange: PropTypes.func,
+    inputRef: PropTypes.func,
+    placeholder: PropTypes.string,
+    maxHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    disabled: PropTypes.bool,
+    options: PropTypes.array.isRequired,
+    noOptionsMessage: PropTypes.string,
+    renderOption: PropTypes.func,
+    renderDropdown: PropTypes.func,
+    optionFilter: PropTypes.func
+  };
+
+  static defaultProps = {
+    noOptionsMessage: 'No matches found',
+    maxHeight: '400px'
+  };
+
+  state = {
+    isOpen: false,
+    isFocused: false,
+    focusedKey: undefined,
+    inputValue: '',
+    value: undefined
+  };
+
+  defaultOptionFilter = (original, comparison) => {
+    const formattedOriginal = original.replace(/ /u, '').toLocaleLowerCase();
+    const formattedComparison = comparison.replace(/ /u, '').toLocaleLowerCase();
+
+    return formattedOriginal.indexOf(formattedComparison) !== -1;
+  };
+
+  getMatchingOptions = () => {
+    const optionFilter = this.props.optionFilter || this.defaultOptionFilter;
+
+    return this.props.options.filter(option => optionFilter(option.label, this.state.inputValue));
+  };
+
+  renderMenuItems = getItemProps => {
+    const { noOptionsMessage, renderOption, selectedValue } = this.props;
+    const { focusedKey } = this.state;
+
+    const matchingOptions = this.getMatchingOptions().map(option => {
+      const checked = selectedValue === option.value;
+
+      const props = getItemProps({
+        key: option.value,
+        focused: focusedKey === option.value,
+        checked,
+        'aria-selected': checked
+      });
+
+      props.children = option.label;
+
+      if (renderOption) {
+        return renderOption(props);
+      }
+
+      return <Item {...props} />;
+    });
+
+    if (matchingOptions.length > 0) {
+      return matchingOptions;
+    }
+
+    return <StyledNoItemsMessage>{noOptionsMessage}</StyledNoItemsMessage>;
+  };
+
+  render() {
+    const {
+      label,
+      'aria-label': ariaLabel,
+      hint,
+      disabled,
+      options,
+      selectedValue,
+      onChange,
+      placeholder,
+      maxHeight,
+      renderDropdown,
+      inputRef,
+      message,
+      small,
+      validation
+    } = this.props;
+    const { isOpen, focusedKey, isFocused, inputValue } = this.state;
+
+    const optionDictionary = options.reduce((dictionary, option) => {
+      dictionary[option.value] = option.label;
+
+      return dictionary;
+    }, {});
+
+    return (
+      <FieldContainer>
+        {({ getLabelProps, getInputProps: getFieldInputProps, getHintProps }) => (
+          <TextGroup>
+            {label && <Label {...getLabelProps()}>{label}</Label>}
```

Comment:
A `Label` click should be focusing the field (it does in the `example/autocomplete.md`).

Reply:
I've updated this as well as handling the `onMouseEnter` and `onMouseLeave` functionality.